presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkSpilledJoinQueries.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+
+import static com.facebook.presto.spark.PrestoSparkQueryRunner.createHivePrestoSparkQueryRunner;
+import static com.facebook.presto.tests.QueryAssertions.assertContains;
+import static io.airlift.tpch.TpchTable.getTables;
+import static org.testng.Assert.assertEquals;
+
+public class TestPrestoSparkSpilledJoinQueries
+        extends TestPrestoSparkJoinQueries
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+    {
+        ImmutableMap.Builder<String, String> configProperties = ImmutableMap.builder();
+        configProperties.put("experimental.spill-enabled", "true");
+        configProperties.put("experimental.spill-in-small-memory-pool", "true");
+        configProperties.put("experimental.join-spill-enabled", "true");
+        configProperties.put("task.concurrency", "2");
+        configProperties.put("experimental.temp-storage-buffer-size", "1MB");
+        configProperties.put("spark.memory-revoking-threshold", "0.0");
+        configProperties.put("experimental.spiller-spill-path", Paths.get(System.getProperty("java.io.tmpdir"), "presto", "spills").toString());
+        return createHivePrestoSparkQueryRunner(getTables(), configProperties.build());
+    }
+
+    @Test
+    public void testJoinWithoutLimit()
+    {
+        assertQuery("SELECT o1.orderkey, o2.orderkey FROM orders o1 JOIN orders o2 on o1.orderkey = o2.orderkey");
+    }
+
+    @Test(enabled = false)
+    @Override
+    public void testLimitWithJoin()
+    {
+        MaterializedResult actual = computeActual("SELECT o1.orderkey, o2.orderkey FROM orders o1 JOIN orders o2 on o1.orderkey = o2.orderkey LIMIT 10");
```

Comment:
I'm a little confused about this test. If the goal is to disable it why is it needed to change the implementation?

Reply:
It narrows down which query caused failure.

Reply:
Do you think leaving a note in a comment section would suffice?

Reply:
+1 I think a comment would be more helpful.  Otherwise it's likely that someone will just switch it to enabled.

Reply:
I would recommend calling `super.testLimitWithJoin()` or simply leaving the method body empty (maybe with only the comment explaining why the test is disabled) instead of copying the entire test implementation (similarly to what we do in other places: https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkAbstractTestQueries.java#L74, https://github.com/prestodb/presto/blob/master/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java#L67)